server/util/auth.js

```diff
@@ -32,24 +32,23 @@ export const authUser = (req, res, next) => {
  * @param {Request|string} req request passed in OR req.headers.authorization (both supported)
  * @returns {Promise<string>}
  */
-export const authByToken = req =>
-  new Promise((resolve, reject) => {
-    let idToken;
-    if (
-      req.headers &&
-      req.headers.authorization &&
-      req.headers.authorization.startsWith('Bearer ')
-    ) {
-      idToken = req.headers.authorization.split('Bearer ')[1];
-    } else if (req.startsWith && req.startsWith('Bearer ')) {
-      idToken = req.split('Bearer ')[1];
-    } else {
-      reject({ token: 'No proper token provided' });
-    }
-    try {
-      const decoded = jwt.verify(idToken, env.jwt);
-      resolve(decoded._id);
-    } catch (err) {
-      reject({ token: 'Invalid token' });
-    }
-  });
+export const authByToken = req => {
+  let idToken;
+  if (
+    req.headers &&
+    req.headers.authorization &&
+    req.headers.authorization.startsWith('Bearer ')
```

Comment:
req?.headers?.authorization?.startsWith && req?.headers?.authorization?.startsWith('Bearer ')

Reply:
At this time, NodeJS not support optional chaining. From what I've seen, however, Node 14 will be supporting this feature, but it will not be out until April.